### DISCUSSION TITLE
I'm trying to use the new JSX implementation but encountered this error for home page. Can anyone help?

### DISCUSSION THREAD

**Discussion by pranavcpawar** [MetaMask/snaps#2604] [Q&A]
![image](https://github.com/user-attachments/assets/a235afb8-4d91-431f-a3a3-0510a56f0291)

**Comment by Montoya** [comment]
Are you using Flask version 12? It hasn't released yet so you would have to get it from a pre-release build.

**Reply by pranavcpawar** [reply]
Thank you @Montoya! I was using Flask version 11

**Reply by pranavcpawar** [reply]
Hey @Montoya, I'm getting the same error again for Flask version 12 but this time only when I nest components.
```
    <Box>
      <Heading>Interactive UI Example Snap</Heading>
      <Form name="example-form">
        <Field label="Example Input">
          <Input name="example-input" placeholder="Enter something..." />
        </Field>
        <Field label="Example Dropdown">
          <Dropdown name="example-dropdown">
            <Option value="option1">Option 1</Option>
            <Option value="option2">Option 2</Option>
            <Option value="option3">Option 3</Option>
          </Dropdown>
        </Field>
        <Field label="Example Checkbox">
          <Checkbox name="example-checkbox" label="Checkbox" />
        </Field>
        <Button type="submit" name="submit">
          Submit
        </Button>
      </Form>
    </Box>
```

**Reply by pranavcpawar** [reply]
```
    <Box>
      <Heading>Interactive UI Example Snap</Heading>
      <Dropdown name="example-dropdown">
        <Option value="option1">Option 1</Option>
        <Option value="option2">Option 2</Option>
        <Option value="option3">Option 3</Option>
      </Dropdown>
    </Box>
```
![image](https://github.com/user-attachments/assets/03d48b38-62e4-4be8-923b-976125274a51)

but fails for above code snippet

**Reply by pranavcpawar** [reply]
Hello again @Montoya, I just figured out that checkbox component is causing the above error. May I know if there is any work around for this?

**Reply by Montoya** [reply]
Hi, I apologize for not seeing your questions before. I would recommend following the code examples here to make sure you have the right syntax: https://docs.metamask.io/snaps/features/custom-ui/with-jsx/

Also, I don't think the Checkbox component is available yet. You should only be using the components on that page ^

**Reply by Montoya** [reply]
@ziad-saab can follow up if you are still stuck

**Reply by ziad-saab** [reply]
Hey @pranavcpawar happy to help! It's been a while since you've chimed in here, so please let me know where you're at and what I can help with.

**Reply by pranavcpawar** [reply]
hey @ziad-saab sir, 
`'Box' cannot be used as a JSX component.
  Its type 'SnapComponent<BoxProps, "Box">' is not a valid JSX element type.
    Type 'SnapComponent<BoxProps, "Box">' is not assignable to type '(props: any, deprecatedLegacyContext?: any) => ReactNode'.
      Type 'SnapElement<BoxProps, "Box">' is not assignable to type 'ReactNode'.
        Property 'children' is missing in type 'SnapElement<BoxProps, "Box">' but required in type 'ReactPortal'.ts(2786)
index.d.ts(378, 9): 'children' is declared here.`
all the JSX components are giving this error prompt but UI is working fine inside snap

**Reply by pranavcpawar** [reply]
hey @ziad-saab sir, the current template-snap-monorepo is bootstrapped with Gatsby. Can u provide a NextJs or Vite implementation of the same?
